pixl_imaging/src/pixl_imaging/_processing.py

```diff
@@ -143,27 +144,112 @@ async def _add_project_to_study(
+      online secondary archive using the study UID. If UID is not available, query on
+      MRN + accession number.
+    - If not found in the online secondary archive, query the nearline secondary archive. If found
+      in the nearline, assume the study is in the primary archive and log it.
+    - If not found in either the primary archive or the online secondary archive,
+      raise a PixlDiscardError.
     """
-    query_id = await orthanc_raw.query_remote(
-        study.orthanc_uid_query_dict, modality=config("VNAQR_MODALITY")
+    query_id = await _find_study_in_archive(
+        orthanc_raw=orthanc_raw,
+        study=study,
+        modality=config("PRIMARY_DICOM_MODALITY"),
     )
-    if query_id is None:
-        logger.info(
-            "No study found with UID {}, trying MRN and accession number", study.message.study_uid
-        )
-        query_id = await orthanc_raw.query_remote(
-            study.orthanc_query_dict, modality=config("VNAQR_MODALITY")
+
+    if query_id is not None:
+        return str(query_id)
+    if _is_daytime() or _is_weekend():
+        msg = (
+            f"Failed to find study {study.message.study_uid} in primary archive. "
+            "Not querying secondary archive during the daytime or on the weekend."
         )
+        raise PixlDiscardError(msg)
+
+    logger.info(
+        "Failed to find study {} in primary archive, trying secondary archive",
+        study.message.study_uid,
+    )
+    query_id = await _find_study_in_archive(
+        study=study,
+        orthanc_raw=orthanc_raw,
+        modality=config("SECONDARY_DICOM_MODALITY"),
+    )
+
     if query_id is None:
-        msg = "Failed to find in the VNA"
+        msg = f"Failed to find study {study.message.study_uid} in primary or secondary archive."
+        raise PixlDiscardError(msg)
+
+    # Check the study is in the online secondary archive
+    query_answers = await orthanc_raw.get_remote_query_answers(query_id)
+    query_answer_content = await orthanc_raw.get_remote_query_answer_content(
+        query_id=query_id,
+        answer_id=query_answers[0],
+    )
+    availablility_tag = "0008,0056"
+    availability = query_answer_content[availablility_tag]["Value"]
+    if availability != "ONLINE":
+        msg = (
+            f"Study {study.message.study_uid} found in {availability} secondary archive "
+            "but we only pull from the online secondary archive."
+        )
         raise PixlDiscardError(msg)
+
     return query_id
 
 
+async def _find_study_in_archive(
+    orthanc_raw: Orthanc,
+    study: ImagingStudy,
+    modality: str,
+) -> Optional[str]:
+    """
+    Query the primary archive for the study using its UID.
+    If UID is not available, query on MRN and accession number.
+    """
+    # We don't want to normalize the query otherwise only MainDicomTags will be returned
+    # (InstanceAvailability will be ignored)
+    additional_data = {"Normalize": False}
```

Comment:
I've not add `"Normalize"` to `ImagingStudy.orthanc_uid_query_dict` as this caused an error with